sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -1149,7 +1149,9 @@ object StateStore extends Logging {
         threadPool.shutdownNow() // Cancel currently executing tasks
 
         // Wait a while for tasks to respond to being cancelled
-        if (!threadPool.awaitTermination(60, TimeUnit.SECONDS)) {
+        // To avoid long test times, use minimum of timeout value or 60 seconds
+        if (!threadPool.awaitTermination(Math.min(60, shutdownTimeout),
```

Comment:
If the timeout is set to be very low - lets say `1s`, we would now wait only for a total of `2s` before we can confirm the tpool shutdown. Should we really repurpose the timeout here ?